executor/analyze.go

```diff
@@ -408,6 +410,41 @@ func (e *AnalyzeExec) analyzeWorker(taskCh <-chan *analyzeTask, resultsCh chan<-
 	}
 }
 
+func analyzeColumnsPushdownWithRetry(e *AnalyzeColumnsExec) *statistics.AnalyzeResults {
+	analyzeResult := analyzeColumnsPushdown(e)
+	// do not retry if succeed / not oom error / not auto-analyze / samplerate not set
+	if analyzeResult.Err == nil || analyzeResult.Err != errAnalyzeOOM ||
+		!e.ctx.GetSessionVars().InRestrictedSQL ||
+		e.analyzePB.ColReq == nil || *e.analyzePB.ColReq.SampleRate <= 0 {
+		return analyzeResult
+	}
```

Comment:
We need to assert statsVersion if we only wanted retry v2 analyze job according to `	// only statistics.Version2 will throw oom error and retry`

Reply:
Updated